src/util/filter.ts

```diff
@@ -0,0 +1,10 @@
+interface Predicate<P> {
+  (item: P, index: number): boolean,
+}
+
+/**
+ * Filter with multiple predicate functions.
+ */
+export const filter = <T>(array: T[], predicateArray: (Predicate<T>)[]) => {
+  return predicateArray.length > 0 ? array.filter((item: T, index) => predicateArray.every(predicate => predicate(item, index))) : array
+}
```

Comment:
This functionality can be achieved more simply with function composition:

```
items.filter(and(predicate1, predicate2, etc))
```

`and` is a basic higher order function. In order to get access to it you can import [fp-and-or](https://github.com/raineorshine/fp-and-or).

Reply:
That makes sense. Thank you!

Reply:
Yeah, super handy. I'm surprised they're not in lodash. I finally made a package just for them since I use them so often :).